diff --git a/lib.commonjs/providers/provider-jsonrpc.js b/lib.commonjs/providers/provider-jsonrpc.js
index a246dca95577b1195d4c8067ddb40540848eb3f9..be22762f5b4126ef9c8d005659b2c4c4c1518a1b 100644
--- a/lib.commonjs/providers/provider-jsonrpc.js
+++ b/lib.commonjs/providers/provider-jsonrpc.js
@@ -644,7 +644,12 @@ class JsonRpcApiProvider extends abstract_provider_js_1.AbstractProvider {
                 operation: payload.method, info: { error }
             });
         }
-        return (0, index_js_5.makeError)("could not coalesce error", "UNKNOWN_ERROR", { error });
+
+        let url = 'unknown';
+        if (this instanceof JsonRpcProvider) {
+            url = this._getConnection().url;
+        }
+        return (0, index_js_5.makeError)(`could not coalesce error: method ${method}, chainId ${this.#network?.chainId}, url ${url}`, "UNKNOWN_ERROR", { error });
     }
     /**
      *  Requests the %%method%% with %%params%% via the JSON-RPC protocol