diff --git a/lib.commonjs/providers/provider-jsonrpc.js b/lib.commonjs/providers/provider-jsonrpc.js
index 05eab9888ace4a4fdd15a3a230c96cf46888b45b..9c4b169c0224bf9384952f71c9c122b3ae0aa39c 100644
--- a/lib.commonjs/providers/provider-jsonrpc.js
+++ b/lib.commonjs/providers/provider-jsonrpc.js
@@ -650,7 +650,12 @@ class JsonRpcApiProvider extends abstract_provider_js_1.AbstractProvider {
                 operation: payload.method, info: { error, payload }
             });
         }
-        return (0, index_js_5.makeError)("could not coalesce error", "UNKNOWN_ERROR", { error, payload });
+
+        let url = 'unknown';
+        if (this instanceof JsonRpcProvider) {
+            url = this._getConnection().url;
+        }
+        return (0, index_js_5.makeError)(`could not coalesce error: method ${method}, chainId ${this.#network?.chainId}, url ${url}`, "UNKNOWN_ERROR", { error });
     }
     /**
      *  Requests the %%method%% with %%params%% via the JSON-RPC protocol